packages/ozone-helper/ozone-type/ozone/model/FlowrWarningLog.ts

```diff
@@ -0,0 +1,65 @@
+import { Instant, Item, OzoneType, UUID } from './Item'
+import { isLocationBuilding, LocationBuilding } from './LocationBuilding'
+import { PersistedFlowrWarningOperator } from './FlowrWarningOperator'
+import { isLocationSite, LocationSite } from './LocationSite'
+
+export enum LOG_ACTION {
+	LOGIN = 'LOGIN',
+	LOGOUT = 'LOGOUT',
+	SWITCH_SITE = 'SWITCH_SITE',
+	WARNING_ON = 'WARNING_ON',
+	WARNING_OFF = 'WARNING_OFF',
+	WARNING_TEST = 'WARNING_TEST'
+}
+
+class BareFlowrWarningLog extends Item {
+	date: Instant
+	action: LOG_ACTION
+	constructor(src: BareFlowrWarningLog) {
+		super(src)
+		this.date = src.date
+		this.action = src.action
+	}
+}
+
+@OzoneType('flowr.warning.log')
+export class FlowrWarningLog extends BareFlowrWarningLog {
+	operator: UUID
+	targetType?: string
+	target?: UUID
+	constructor(src: FlowrWarningLog) {
+		super(src)
+		this.operator = src.operator
+		this.targetType = src.targetType
+		this.target = src.target
+	}
+}
+
+export class HydratedFlowrWarningLog extends BareFlowrWarningLog {
+	id: UUID
```

Comment:
Item already has an id property so I think this override is unecessary

Reply:
this one is non nullable